core/src/main/java/org/projectnessie/cel/interpreter/AttributeFactory.java

```diff
@@ -353,34 +354,35 @@ public Object resolve(org.projectnessie.cel.interpreter.Activation vars) {
      * type, then the result is `nil`, `false`, `nil` per the interface requirement.
      */
     @Override
-    public Object tryResolve(org.projectnessie.cel.interpreter.Activation vars) {
+    public Pair<Object, Boolean> tryResolve(org.projectnessie.cel.interpreter.Activation vars) {
       for (String nm : namespaceNames) {
         // If the variable is found, process it. Otherwise, wait until the checks to
         // determine whether the type is unknown before returning.
-        Object op = vars.resolveName(nm);
-        if (op != null) {
+        Pair<Object, Boolean> obj = vars.resolveName(nm);
+        if (obj.b) {
+          Object op = obj.a;
           for (Qualifier qual : qualifiers) {
             Object op2 = qual.qualify(vars, op);
             if (op2 instanceof Err) {
-              return op2;
+              return new Pair<>(op2, true);
             }
             if (op2 == null) {
               break;
             }
             op = op2;
           }
-          return op;
+          return new Pair<>(op, true);
         }
         // Attempt to resolve the qualified type name if the name is not a variable identifier.
         Val typ = provider.findIdent(nm);
         if (typ != null) {
           if (qualifiers.isEmpty()) {
-            return typ;
+            return new Pair<>(typ, true);
           }
           throw noSuchAttributeException(this);
         }
       }
-      return null;
+      return new Pair<>(null, false);
```

Comment:
An an alternative, why not fail here (`throw noSuchAttributeException(this)`)? This way, `tryResolve` either succeeds (returning a potentially null value) or fails (throwing an exception). Then `resolve` would need the following adjustments:

```java
    @Override
    public Object resolve(org.projectnessie.cel.interpreter.Activation vars) {
        for (NamespacedAttribute attr : attrs) {
          try {
            return attr.tryResolve(vars);
          } catch (Exception e) {}
        }
       throw noSuchAttributeException(this);
   }
```

Reply:
Yeah, there are actually 2 interfaces and I agree that `tryResolve` could be avoided with 1 try catch and exception ignoring.

Another interface is `resolveName`, which I'd rather leave with `Pair<Object, Boolean>` implementation, 'cause it will require put `findIdent` logic inside `catch` section, which could be confusing🤷.

Here's example:
```java
@Override
    public Object tryResolve(org.projectnessie.cel.interpreter.Activation vars) {
      for (String nm : namespaceNames) {
        // If the variable is found, process it. Otherwise, wait until the checks to
        // determine whether the type is unknown before returning.
       try {
           Object op = vars.resolveName(nm);
           for (Qualifier qual : qualifiers) {
             Object op2 = qual.qualify(vars, op);
             if (op2 instanceof Err) {
               return op2;
             }
             if (op2 == null) {
               break;
             }
             op = op2;
           }
           return op;
         }
       } catch (Exception ignore) {
           // Attempt to resolve the qualified type name if the name is not a variable identifier.
           Val typ = provider.findIdent(nm);
           if (typ != null) {
             if (qualifiers.isEmpty()) {
               return typ;
             }
             throw noSuchAttributeException(this);
          }
         }
       }
      throw noSuchAttributeException(this);
    }
``` 

I'm gonna change `tryResolve` with `try catch` approach, it looks really cleaner. 

If  `resolveName` with exception throwing is more preferable, I can change that too :)